control-plane/connect-inject/controllers/endpointsv2/endpoints_controller.go

```diff
@@ -510,6 +545,22 @@ func (r *Controller) getConsulPartition() string {
 	return r.ConsulPartition
 }
 
+// getWriteCacheKey gets a key to track syncronization of a K8s service to deduplicate writes to Consul.
+func getWriteCacheKey(serviceName types.NamespacedName) string {
+	// The only value we have to rely on when deleting from the cache is the namespaced name (not the UID).
+	// We aggressively delete from the cache whenever an invalidating event occurs, and rely on writes
+	// for eventual consistency. See also WriteCache.hasMatch.
+	return serviceName.String()
+}
+
+// getWriteHash gets a hash of the given resource to deduplicate writes to Consul.
+// This hash is not intended to be cryptographically secure, only deterministic and collision-resistent for tens of
+// thousands of values.
+func getWriteHash(resource *pbresource.Resource) []byte {
+	h := sha256.Sum256(resource.Data.Value)
```

Comment:
I wonder if encoding an `any.Any` field into `[]byte` will do a plain Marshal or the Deterministic=true one: https://pkg.go.dev/google.golang.org/protobuf/proto#MarshalOptions

Reply:
See my other comment.

Reply:
Yep, I think that was the right fix you suggested - that way we avoid deterministically encoding a non-deterministic payload 😅